### PR TITLE
Fix handling of trailers in relay when chunking

### PR DESCRIPTION
**Pull request by ferstlf**
The relay client splits responses from the backend into multiple chunks based on MaxChunkSize and sends these chunks to the relay server. Currently, the relay client includes HTTP trailers in the last chunk sent - which makes sense - but the relay server code does not correctly handle the case with more than one chunk and always returns the trailers sent with the first chunk (ignoring the trailers in all other chunks). Thus when chunking responses we currently do not return any trailers. 

This PR fixes this issue in the relay server by handling trailers similar to the message body and collecting all trailers from all chunks before appending them to the HTTP response sent to the user client. We add an integration test and unit tests that cover chunked responses.

This fixes the issue that relaying gRPC requests with responses larger than MaxChunkSize leads to a `UNKNOWN` (`No status received`) error. This also removes an existing [TODO](https://github.com/googlecloudrobotics/core/blob/63b665ba7a0a9d9e46d0e387397ceade6db6659e/src/go/cmd/http-relay-server/server/server.go#L163) which has spotted exactly this issue before.

### CODE REVIEW

File: src/go/cmd/http-relay-server/server/server.go [hunk 1]
```diff
@@ -129,40 +129,55 @@ func extractBackendNameAndPath(r *http.Request) (backendName string, path string
 	return
 }
 
+type responseChunk struct {
+    Body []byte
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/server/server.go:133]: please indent

File: src/go/cmd/http-relay-server/server/server.go [hunk 2]
```diff
@@ -129,40 +129,55 @@ func extractBackendNameAndPath(r *http.Request) (backendName string, path string
...
+}
+
 // responseFilter enforces that there's at least one HttpResponse in the out
-// channel and that the first response has a status code. From the reposnses it
-// extracts and return headers, trailers, status-code and body data.
-func responseFilter(backendCtx backendContext, in <-chan *pb.HttpResponse) ([]*pb.HttpHeader, []*pb.HttpHeader, int, <-chan []byte) {
-	body := make(chan []byte, 1)
+// channel and that the first response has a status code. From the responses it
+// extracts it synchronously returns headers and status-code and asynchronously
+// returns body and trailers via a channel.
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/server/server.go:140]: I think there is something wrong in the comment

File: src/go/cmd/http-relay-server/server/server.go [hunk 3]
```diff
@@ -129,40 +129,55 @@ func extractBackendNameAndPath(r *http.Request) (backendName string, path string
...
 		// Flush remaining messages
 		for range in {
 		}
-		return nil, nil, http.StatusInternalServerError, body
+		return nil, http.StatusInternalServerError, responseChunks
 	}
-	body <- []byte(firstMessage.Body)
-	lastMessage := firstMessage
+
+    responseChunks <- &responseChunk{
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/server/server.go:165]: indent

File: src/go/cmd/http-relay-server/server/server.go (hunk 1 above)
Comment (ferstlf) [reply src/go/cmd/http-relay-server/server/server.go]: You were too quick with your review :) Now all files are auto-formatted.

**Review by ensonic — APPROVED** [review]
Thanks

**Pull request merged**